sklearn/utils/_metadata_requests.py

```diff
@@ -434,12 +440,28 @@ def _route_params(self, params):
             elif alias in args:
                 res[prop] = args[alias]
         if unrequested:
+            if parent.__class__.__name__ != "str":
```

Comment:
This should be documented in the docstring. Do you have a concrete example where `parent` is a string. Also what is the problem in this case?

Reply:
Now this distinction is not necessary anymore, since all MetadataRouter.owner is always a string.